pkg/sdk/errors.go

```diff
@@ -48,3 +47,230 @@ func decodeDriverError(err error) error {
 
 	return err
 }
+
+const ghIssueBodyTemplate = `
+**Provider Version**
+
+The provider version you are using.
+
+**Terraform Version**
+
+The version of Terraform you were using when the bug was encountered.
+
+**Describe the bug**
+
+A clear and concise description of what the bug is.
+
+**Expected behavior**
+
+A clear and concise description of what you expected to happen.
+
+**Code samples and commands**
+
+Please add code examples and commands that were run to cause the problem.
+
+**Additional context**
+
+Add any other context about the problem here.
+
+<!-- 
+Please provide additional error messages if we missed any (see the errors below and compare it with your console output)
+-->
+
+Errors:
+%s
+`
+
+// NewTopLevelError wraps an error with "final" error message of sdk.
+// It should be placed in the highest place of call stack to catch as much error context as possible.
+// TODO Is it possible to call this function somewhere high in the callstack, so it always be top level ? If not maybe this would be overkill / too hard to use
+// TODO It's possible to wrap errors multiple times, this function will try to keep only the one with most error context.
```

Comment:
If confusing this TODO and the one under is referring following case
err (type unknown error)
|
- err1 (type SDKError)
- err2 (type TopLevelError)
|
- err3 (type SDKError)

err has child err2 which is TopLevelError, we want to unwrap it and wrap err, so the error will have more context (TopLevelError now will also include err1) and we should end up with

err (TopLevelError)
|
- err1 (SDKError)
- err3 (SDKError

Reply:
You can add this comment to the comment in code, to make it understandable also from there.